### PR TITLE
chore(common): CHECKOUT-000 Upgrade to typescript version 4.1

### PR DESCRIPTION
**Pull request by animesh1987**
## What?
- Upgrade to typescript 4.1
- Upgrade typedoc and typedoc-plugin-markdown to compatible versions and update docs command

## Why?
As part of working with Nx there are few features that require typescript version 4+. Plus we get better type checks plus the present version is fairly dated. 

## Testing / Proof
- Circle

@bigcommerce/checkout

### CODE REVIEW

**Review by davidchin — APPROVED** [review]
One small comment. Otherwise LGTM. Thanks for doing this, one step closer to the latest TS version!

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "typescript.tsdk": "node_modules/typescript/lib"
```
Comment (bc-peng) [inline .vscode/settings.json:2]: Think we'd better add `.vscode` and `.idea` to SDK's `.gitignore` as well. 🙇🏻

[the remark this answers is not in the document]
Comment (animesh1987) [reply .vscode/settings.json:2]: Updated as @bc-peng suggestion added them to `.gitignore`

Comment (animesh1987) [reply .vscode/settings.json:2]: Done

**Pull request merged**